services/vision/server.go

```diff
@@ -239,6 +223,84 @@ func (server *serviceServer) GetProperties(ctx context.Context,
 	return out, nil
 }
 
+func (server *serviceServer) CaptureAllFromCamera(
+	ctx context.Context,
+	req *pb.CaptureAllFromCameraRequest,
+) (*pb.CaptureAllFromCameraResponse, error) {
+	ctx, span := trace.StartSpan(ctx, "service::vision::server::CaptureAllFromCamera")
+	defer span.End()
+	svc, err := server.coll.Resource(req.Name)
+	if err != nil {
+		return nil, err
+	}
+	captOptions := viscapture.CaptureOptions{
+		ReturnImage:           req.ReturnImage,
+		ReturnDetections:      req.ReturnDetections,
+		ReturnClassifications: req.ReturnClassifications,
+		ReturnObject:          req.ReturnObjectPointClouds,
+	}
+	capt, err := svc.CaptureAllFromCamera(ctx,
+		req.CameraName,
+		captOptions,
+		req.Extra.AsMap(),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	objProto, err := segmentsToProto(req.CameraName, capt.Objects)
+	if err != nil {
+		return nil, err
+	}
+
+	imgProto, err := imageToProto(ctx, capt.Image, req.CameraName)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pb.CaptureAllFromCameraResponse{
+		Image:           imgProto,
+		Detections:      detsToProto(capt.Detections),
+		Classifications: clasToProto(capt.Classifications),
+		Objects:         objProto,
+	}, nil
+}
+
+func imageToProto(ctx context.Context, img image.Image, cameraName string) (*camerapb.Image, error) {
+	if img == nil {
+		return &camerapb.Image{}, nil
+	}
+	imgBytes, mimeType, err := encodeUnknownType(ctx, img, utils.MimeTypeJPEG)
+	if err != nil {
+		return nil, err
+	}
+	format := utils.MimeTypeToFormat[mimeType]
+	return &camerapb.Image{
+		Image:      imgBytes,
+		Format:     format,
+		SourceName: cameraName,
```

Comment:
it seems that `SourceName`  gets lost when the proto is decoded in the `image.Image`....

Reply:
luckily, SourceName is only really important for GetImages -- so I think this is fine